### PR TITLE
Ability to set unlimited bad tolerance in import tool

### PR DESCRIPTION
**Pull request by tinwelint**
by using --bad-tolerance -

### CODE REVIEW

**Comment by spacecowboy** [comment]
Nice, then I didn't have to do it all myself (see https://github.com/neo4j/neo4j/pull/8968)

But why did you pick `-` as the infinity value? I think that might be confusing. A naked dash is typically used to indicate that stdin should be read.

for example:
http://unix.stackexchange.com/questions/41828/what-does-dash-at-the-end-of-a-command-mean

which means some users could be confused by `neo4j-import --bad-tolerance -`

**Comment by tinwelint** [comment]
@spacecowboy right you are, what would be better?

**Comment by spacecowboy** [comment]
@tinwelint `-1` comes to mind, it also matches what the constant is. otherwise, a string matching a positive action like `true` perhaps

**Comment by tinwelint** [comment]
Good idea, I'd say `true` in that case. `-1` feels like a developer thing, not a user thing to me.

**Comment by burqen** [comment]
@tinwelint Can we have some labels on this guy please.

**Comment by tinwelint** [comment]
@spacecowboy did you already merge related stuff to this commit?

**Comment by spacecowboy** [comment]
@tinwelint I have made the almost equivalent changes for `neo4j-admin` on 3.2. It contains a subset of your commit and should not conflict (git will complain about a conflict but the code should be identical).

Ref: https://github.com/neo4j/neo4j/pull/8968
subset of this pr: https://github.com/neo4j/neo4j/pull/8968/commits/1e313023128d8459169251b38c7f052c1d30cb03

File: community/import-tool/src/main/java/org/neo4j/tooling/ImportTool.java [hunk 1]
```diff
@@ -99,6 +100,7 @@
  */
 public class ImportTool
 {
+    private static final String UNLIMITED = "-";
```
Comment (MishaDemianenko) [inline community/import-tool/src/main/java/org/neo4j/tooling/ImportTool.java]: As @spacecowboy already mentioned that would be much more understandable to use `-1` as representation of unlimited value

File: community/import-tool/src/test/java/org/neo4j/tooling/ImportToolTest.java [hunk 2]
```diff
@@ -1138,6 +1138,27 @@ public void shouldPrintReferenceLinkOnDataImportErrors() throws Exception
                 "/import/import-tool-header-format/#import-tool-id-spaces" );
     }
 
+    @Test
+    public void shouldCollectUnlimitedNumberOfBadEntries() throws Exception
+    {
+        // GIVEN
+        List<String> nodeIds = new ArrayList<>();
```
Comment (MishaDemianenko) [inline community/import-tool/src/test/java/org/neo4j/tooling/ImportToolTest.java]: will `Collections.nCopies( 10_000, "A" )` do the trick instead?

**Review by MishaDemianenko — CHANGES_REQUESTED** [review]
One usability and minor comment, good otherwise

File: community/import-tool/src/main/java/org/neo4j/tooling/ImportTool.java (hunk 1 above)
Comment (tinwelint) [reply community/import-tool/src/main/java/org/neo4j/tooling/ImportTool.java]: I do think -1 isn't particularly user friendly, is it? I'd vote for `true`, as discussed above

Comment (MishaDemianenko) [reply community/import-tool/src/main/java/org/neo4j/tooling/ImportTool.java]: do you mean `neo4j-import --bad-tolerance -unlimited=true` or smt else?

Comment (tinwelint) [reply community/import-tool/src/main/java/org/neo4j/tooling/ImportTool.java]: `--bad-tolerance=true`

I think `--bad-tolerance=-1` is so developer:y

**Review by MishaDemianenko — APPROVED** [review]

**Pull request merged**